src/core/src/bootstrap/EnvLayer.py

```diff
@@ -64,20 +64,25 @@ def get_package_manager(self):
         """ Detects package manager type """
         ret = None
 
-        # choose default - almost surely one will match.
-        for b in ('apt-get', 'yum', 'zypper'):
-            code, out = self.run_command_output('which ' + b, False, False)
+        if self.platform.linux_distribution()[0] == Constants.AZURE_LINUX:
```

Comment:
This can dangerously break execution as an unprotected general code path because:
![image](https://github.com/user-attachments/assets/f1daf337-b9a7-49eb-94b9-fa3a45894cbf)

Line 287 in EnvLayer is not safe due to the deprecation - affects other code paths too.

Not a specific-to-this-PR concern but needs to be addressed for the code base.

Reply:
I'm confused.. Haven't we already addressed this by checking the python version?

<img width="400" alt="image" src="https://github.com/user-attachments/assets/33ed2f49-cdb8-4201-aae5-8182d1acc4e9" />

Reply:
The background is that even distro.linux_distribution is deprecated. But we're not immdidately affected by that as we have a static copy held with us. Non-ideal since we'll get hit with issues with stale code, but there will be future work to update the dependency and then move to distro.name or similar.